src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -497,13 +497,13 @@ private BoundExpression MakeCallWithNoExplicitArgument(MethodArgumentInfo method
             if (method.IsExtensionMethod)
             {
                 Debug.Assert(expression == null);
-                Debug.Assert(method.Parameters.AsSpan()[1..].All(assertParametersAreOptional, (p, assertOptional) => (p.IsOptional || p.IsParams || !assertOptional) && p.RefKind == RefKind.None));
-                Debug.Assert(method.ParameterRefKinds.IsDefaultOrEmpty || method.ParameterRefKinds[0] is RefKind.In or RefKind.RefReadOnlyParameter or RefKind.None);
+                Debug.Assert(method.Parameters.AsSpan()[1..].All(assertParametersAreOptional, static (p, assertOptional) => (p.IsOptional || p.IsParams || !assertOptional) && p.RefKind is RefKind.None or RefKind.In or RefKind.RefReadOnlyParameter));
```

Comment:
>or RefKind.RefReadOnlyParameter [](http://example.com/codeflow?start=215&length=31)

Do we really want to allow `ref readonly` parameters here? We will not be able to pass a reference to a "real" location to the call

Reply:
> We will not be able to pass a reference to a "real" location to the call

We should probably warn like we normally do when someone passes an rvalue into a ref readonly parameter. Would prefer to do that in a follow up PR, as this one is just fixing asserts, not changing behavior of the compiler in Release builds.